examples/cloud-operations/network-dashboard/cloud-function/main.py

```diff
@@ -0,0 +1,791 @@
+#
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from google.cloud import monitoring_v3
+from googleapiclient import discovery
+from google.api import metric_pb2 as ga_metric
+import time
+import os
+import google.api_core
+import re
+import random
+
+# 
+monitored_projects_list = os.environ.get("monitored_projects_list").split(",")  # list of projects from which function will get quotas information
+monitoring_project_id = os.environ.get("monitoring_project_id")  # project where the metrics and dahsboards will be created
+monitoring_project_link = f"projects/{monitoring_project_id}"
+service = discovery.build('compute', 'v1')
+
+# DEFAULT LIMITS:
+limit_vpc_peer = os.environ.get("LIMIT_VPC_PEER").split(",")
+limit_l4 = os.environ.get("LIMIT_L4").split(",")
+limit_l7 = os.environ.get("LIMIT_L7").split(",")
+limit_instances = os.environ.get("LIMIT_INSTANCES").split(",")
+limit_instances_ppg = os.environ.get("LIMIT_INSTANCES_PPG").split(",")
+limit_subnets = os.environ.get("LIMIT_SUBNETS").split(",")
+limit_l4_ppg = os.environ.get("LIMIT_L4_PPG").split(",")
+limit_l7_ppg = os.environ.get("LIMIT_L7_PPG").split(",")
+
+def quotas(request):
+  '''
+    Cloud Function Entry point, called by the scheduler.
+
+      Parameters:
+        request: for now, the Cloud Function is triggered by an HTTP trigger and this request correspond to the HTTP triggering request.
+      Returns:
+        'Function executed successfully'
+  '''
+  global client, interval
```

Comment:
Globals are an antipattern in Python. What you probably want is to have a `get_client` function, and each individual function to accept an optional `client` arg.